portal/views.py

```diff
@@ -0,0 +1,174 @@
+        return Response(self.serializer_class(Poll.objects.filter(approved=False), many=True).data)
+
+
+class RetrievePollVotes(APIView):
+    """Retrieve history of polls and their statistics"""
+
+    def get(self, request):
+        if not request.user.is_authenticated:
+            raise PermissionDenied()
+        # filters for all polls that user voted in
+        serializer = RetrievePollVoteSerializer(
+            PollVote.objects.filter(user=request.user), many=True
+        )
+        history_list = serializer.data
+        # filters for all polls user has not voted in
+        remaining_list = RetrievePollSerializer(
+            Poll.objects.filter(expire_date__lte=timezone.localtime()).exclude(
+                id__in=PollVote.objects.filter(user=request.user).values_list("poll", flat=True),
+                approved=True,
+            ),
+            many=True,
+        ).data
+        # puts remaining_list in the same format of history_list, then appends them
+        for entry in remaining_list:
+            context = {"id": entry["id"], "poll": entry, "poll_option": {}}
+            history_list.append(context)
+        # calculates statistics for each poll
+        for entry in history_list:
+            context = {"poll_statistics": self.calculate_statistics(entry["poll"]["id"])}  # stat
+            entry.update(context)
+        return Response(sorted(history_list, key=lambda i: i["poll"]["expire_date"], reverse=True))
+
+    def calculate_statistics(self, id):
+        """Returns the statistics of school and year for each poll"""
+        poll = Poll.objects.get(id=id)
+        options = PollOption.objects.filter(poll=poll)
+        statistic_list = []
+        # loops through each option and aggregates votes
+        for option in options:
+            votes = PollVote.objects.filter(poll_option=option)
+            context = {"schools": {}, "years": {}}
+            for vote in votes:
+                year = 2024  # arbitrary for now, don't know how to get this yet
+                school = self.get_affiliation(vote.user.email)
+                if year not in context["years"].keys():
+                    context["years"][year] = 1
+                else:
+                    context["years"][year] += 1
+                if school not in context["schools"].keys():
+                    context["schools"][school] = 1
+                else:
+                    context["schools"][school] += 1
+            statistic_list.append({option.choice: context})
+        return statistic_list
+
+    def get_affiliation(self, email):
+        """Gets the school based on user's email"""
+
+        if "wharton" in email:
+            return "Wharton"
+        elif "seas" in email:
+            return "SEAS"
+        elif "sas" in email:
+            return "SAS"
+        elif "nursing" in email:
+            return "Nursing"
+        else:
+            return "Other"
+
+
+class CreatePoll(generics.CreateAPIView):
```

Comment:
I'm pretty sure most of these Create/Update/List/ individual views can be merged into viewsets to keep things a bit more concise